contrib/initramfs/zfsunlock

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+set -eu
+echo "Waiting for zfs init script to complete, press Ctrl-C to exit into shell ..."
+while [ ! -e /run/zfs_fs_name ] && [ ! -e /run/zfs_unlock_complete ]; do
+	sleep 1;
```

Comment:
no semicolon needed here or in the sleep below

Reply:
I understand more what you're doing now. You're waiting as long as _neither_ of these exist. But you could still do this as:

```
while [ ! -e /run/zfs_fs_name ]; do
    if [ -e /run/zfs_unlock_complete ]; then
	exit 0
    fi
    sleep 1
done
```
which would have the same effects as now: If `/run/zfs_unlock_complete` shows up, you exit. If `/run/zfs_fs_name` shows up but not `/run/zfs_unlock_complete`, you break out of the loop.

If the filesystem is unencrypted, then `/run/zfs_fs_name` will never be created. Now that `zfsunlock` is run manually, that likely doesn't matter. But even if someone does run it, the check for `/run/zfs_unlock_complete` is sufficient.